pyproject.toml

```diff
@@ -1,16 +1,6 @@
 [build-system]
 requires = ["setuptools",
             "wheel",
-            "cython>=3.0.0",
-            "numpy==1.16.0; python_version <= '3.7'",
-            "numpy==1.17.3; python_version == '3.8'",
-            "numpy==1.19.3; python_version == '3.9'",
-            "numpy==1.21.4; python_version =='3.10'",
-            "numpy==1.23.4; python_version=='3.11'",
-            "numpy; python_version >= '3.12'",
+            "cython>=0.29.21",
+            "numpy>=2.0.0",
```

Comment:
Should we change PR title to indicate that this now makes pycbc require numpy 2 as well as dropping python versions?

Reply:
Is it actually necessary to include Numpy 2? As far as I understood it was supposed to maintain some level of backward compatibility.

Reply:
We don't require numpy 2.0. However, to work with 2.0 it *must* be built with 2.0. They have added in 2.0 the ability for ABI to be backwards compatible, so when  it is built with 2.0 it will still work with 1.XX. 

See 

https://numpy.org/devdocs/dev/depending_on_numpy.html#numpy-2-0-specific-advice

That's why i did not update the setup.py or requirements file at hte same time, only the pyproject.toml which dictates how the build itself is done.